bittide/src/Bittide/CaptureUgn.hs

```diff
@@ -0,0 +1,52 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+module Bittide.CaptureUgn (captureUgn) where
+
+import Clash.Explicit.Prelude hiding (PeriodToCycles)
+
+import Data.Maybe (fromMaybe, isJust)
+import Data.Tuple (swap)
+
+import Protocols
+import Protocols.Wishbone
+
+import Bittide.SharedTypes (Bytes)
+import Bittide.Wishbone (wbToVec)
+
+import qualified Clash.Prelude as C
+
+{- | Capture the UGN from the received data stream. The first `Just` value in the
+elastic buffer contains the rxCounter. Subtracting that from the txCounter results
+in the UGN. All subsequent values in the elastic buffer are bittide data.
```

Comment:
Change to something along the lines of:

```
Captures the remote counter from a bittide link and pairs it with the corresponding local counter.
All frames except for the first frame will be forwarded to the output.

Assumes that:
* When a link is not up, it produces `Nothing` consistently
* When a link comes down, it will at some point switch from only producing `Nothing`, to only producing `Just data`
* When a link comes up, the first frame will contain the remote counter value.
* When a link goes down, it will start producing consistently `Nothing`
```